Error "implicit declaration of function ‘_aligned_malloc’" - swap header order due to MinGW bug?

First of all, thank you for llama.cpp! 

I wanted to build binaries for Windows and I'm getting an error
> /workspace/srcdir/llama.cpp/ggml.c:193:35: error: implicit declaration of function ‘_aligned_malloc’; did you mean

It seems to be linked to an old [MinGW bug](https://sourceforge.net/p/mingw-w64/bugs/192/).
The workaround seems to be just moving inclusion of `stdlib.h` to the top, so it's called before `intrin.h` in the file ggml.c.
In practice, it would have to be [before "ggml-impl.h"](https://github.com/ggerganov/llama.cpp/blob/39d8bc71edcb8b6f99d46fa4216af7a15232e218/ggml.c#L4
), because that calls `intrih.h`.

This workaround seems to be covered in many places, eg, https://stackoverflow.com/a/10862288.

**Steps to reproduce**:

```
cmake .. \
    -DCMAKE_INSTALL_PREFIX=$prefix \
    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
    -DCMAKE_BUILD_TYPE=RELEASE \
    -DBUILD_SHARED_LIBS=ON \
    -DLLAMA_BUILD_TESTS=OFF \
    -DLLAMA_BUILD_EXAMPLES=ON \
    -DLLAMA_NATIVE=OFF \
    -DLLAMA_ACCELERATE=ON \
    -DLLAMA_AVX=ON \
    -DLLAMA_AVX2=ON \
    -DLLAMA_F16C=ON \
    -DLLAMA_FMA=ON \
    -DLLAMA_BLAS=OFF \
    -DLLAMA_CUBLAS=OFF \
    -DLLAMA_CLBLAST=OFF \
```
**Error**:

>   | [12:08:21] /workspace/srcdir/llama.cpp/ggml.c: In function ‘ggml_init’:
>   | [12:08:21] /workspace/srcdir/llama.cpp/ggml.c:193:35: error: implicit declaration of function ‘_aligned_malloc’; did you mean ‘_aligned_realloc’? [-Werror=implicit-function-declaration]


**Version**: Latest master (b1730)
**System**: build system for x86_64 (x86_64-w64-mingw32-gcc (GCC) 10.2.0, clang version 16.0.6)
**Build log**: https://buildkite.com/julialang/yggdrasil/builds/7432#018cba9c-3aff-49f1-a5e3-71ae130b1364

EDIT: The linked build log is for version 1703, but the error persists even for the latest 1730 (tested locally).

--- a/ggml-impl.h
+++ b/ggml-impl.h
@@ -5,6 +5,7 @@
 // GGML internal header
 
 #include <assert.h>
+#include <stdlib.h>
 #include <stddef.h>
 #include <stdbool.h>
 #include <string.h> // memcpy

--- a/ggml-impl.h
+++ b/ggml-impl.h
@@ -5,7 +5,7 @@
 // GGML internal header
 
 #include <assert.h>
-#include <stdlib.h>
+#include <stdlib.h> // load `stdlib.h` before other headers to work around MinGW bug: https://sourceforge.net/p/mingw-w64/bugs/192/
 #include <stddef.h>
 #include <stdbool.h>
 #include <string.h> // memcpy

